test/integration/test_celery_user_rate_limit.py

```diff
@@ -0,0 +1,169 @@
+import datetime
+import tempfile
+from functools import lru_cache
+from typing import (
+    Dict,
+    Iterable,
+    List,
+)
+
+from celery.result import AsyncResult
+from sqlalchemy import text
+
+from galaxy.celery import galaxy_task
+from galaxy.model.database_utils import sqlalchemy_engine
+from galaxy.util import ExecutionTimer
+from galaxy_test.driver.driver_util import init_database
+from galaxy_test.driver.integration_util import IntegrationTestCase
+
+
+@galaxy_task(bind=True)
+def mock_user_id_task(self, task_user_id: int):
+    return task_user_id
+
+
+@lru_cache()
+def sqlite_url():
+    path = tempfile.NamedTemporaryFile().name
+    dburl = f"sqlite:///{path}"
+    init_database(dburl)
+    return dburl
+
+
+@lru_cache()
+def setup_users(dburl: str, num_users: int = 2):
```

Comment:
If the sole purpose of this function is to ensure we have enough users in the test db (currently, only +1 with `id=2`), we already have a `_setup_user` method that sets up a proper user. And if you only need just a row in the user table, you could you a session to simplify this to:
```py
for _ in range(num_users):
    session.add(User(whatever-args-you-need))
session.commit()
```
You don't need to explicitly set the user IDs, right?

Reply:
Having explicitly set user id's of 1 and 2 makes the testing code easier to write. However, if it is important to do so I could change the test code to lookup the user id's. In that case I would just need to ensure that there are at least 2 users defined in the user table.

Reply:
I don't think we can assume specific ID values here: the database may be shared with other tests (a trade-off we accept for the sake of faster tests). We only know for sure that there is a user.id=1; any other User records may, potentially, interfere with the correctness of these tests. Ideally, the test would create the users it needs.

Reply:
The test does not assume the presence of any user other than user id 1. It creates the other users it needs with id's 2 and up. So I don't see any problem here.

Reply:
we cannot assume that user.id=2 (or any other value) is not already taken (by another test).

Reply:
There is no problem if user.id=2 or any other user id is taken up by another test. I just need there to be a record in that table. If it already exists I don't create it. The column values don't matter and I don't manipulate the values. I really don't see what the issue is here. It only needs to exist for the postgres case because of a foreign key constraint from the new celery_user_rate_limit table.

Reply:
Oh, i see - the set difference will exclude any existing ids even if you start with 2. You're right, there's no issue here.